File: problemsets/Codeforces/Python/A994.py
"""
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 """
i=lambda:input().split()
n,m=map(int,i())
a,b=i(),i()
print(*[x for x in a if x in b])

File: problemsets/Codeforces/Javascript/A994.js
/**
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 */
readline()
a = readline().split(' ').map(Number)
b = new Set(readline().split(' ').map(Number))
print(a.filter(x=>b.has(x)).join(" "))